Builds/Test.py

```diff
@@ -75,55 +95,69 @@
     '--keep_going', '-k',
     action='store_true',
     help='Keep going after one configuration has failed.',
- )
+)
 
 parser.add_argument(
     '--silent', '-s',
     action='store_true',
     help='Silence all messages except errors',
- )
+)
 
 parser.add_argument(
     '--verbose', '-v',
     action='store_true',
     help=('Report more information about which commands are executed and the '
           'results.'),
- )
+)
 
 parser.add_argument(
     '--test', '-t',
     default='',
     help='Add a prefix for unit tests',
- )
+)
+
+parser.add_argument(
+    '--nonpm', '-n',
+    action='store_true',
+    help='Do not run npm tests',
+)
+
+parser.add_argument(
+    '--clean', '-c',
+    action='store_true',
+    help='delete all build artifacts after testing',
+)
 
 parser.add_argument(
     'scons_args',
     default=(),
     nargs='*'
-    )
+)
 
 ARGS = parser.parse_args()
 
-def shell(*cmd, **kwds):
-    "Execute a shell command and return the output."
-    silent = kwds.pop('silent', ARGS.silent)
-    verbose = not silent and kwds.pop('verbose', ARGS.verbose)
+
+def shell(cmd, args=(), silent=False):
+    """"Execute a shell command and return the output."""
+    silent = ARGS.silent or silent
+    verbose = not silent and ARGS.verbose
     if verbose:
-        print('$', ' '.join(cmd))
-    kwds['shell'] = IS_WINDOWS
+        print('$' + cmd, *args)
+
+    command = (cmd,) + args
 
     process = subprocess.Popen(
-        cmd,
+        command,
         stdin=subprocess.PIPE,
         stdout=subprocess.PIPE,
         stderr=subprocess.STDOUT,
-        **kwds)
+        shell=IS_WINDOWS)
     lines = []
     count = 0
     for line in process.stdout:
-        lines.append(line)
+        lines.append(line.decode())
         if verbose:
-            print(line, end='')
+            print(line.decode(), end='')
```

Comment:
The obvious difference is that I'm using Python 2.7.8 (from https://www.python.org/downloads/). Windows 7 SP1. Here's the error:
```
Traceback (most recent call last):
  File "./Builds/Test.py", line 265, in <module>
    main()
  File "./Builds/Test.py", line 247, in main
    failed = run_tests(args)
  File "./Builds/Test.py", line 202, in run_tests
    resultcode, lines = shell('npm', ('test', '--rippled=' + executable,))
  File "./Builds/Test.py", line 158, in shell
    lines.append(line.decode())
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 5: ordinal not in range(128)
```
And here's what fixed it for me:
```
index 071d641..c4d9541 100755
--- a/Builds/Test.py
+++ b/Builds/Test.py
@@ -155,9 +155,13 @@ def shell(cmd, args=(), silent=False):
     lines = []
     count = 0
     for line in process.stdout:
-        lines.append(line.decode())
+        if isinstance(line, str):
+            decoded = line
+        else:
+            decoded = line.decode()
+        lines.append(decoded)
         if verbose:
-            print(line.decode(), end='')
+            print(decoded, end='')
         elif not silent:
             count += 1
             if count >= 80:
```